### PR TITLE
Add more metadata to the user-agent string

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: libfwupd/fwupd-common.c [hunk 1]
```diff
@@ -133,3 +135,115 @@ fwupd_checksum_get_best (GPtrArray *checksums)
...
+fwupd_build_user_agent_os_release (void)
+{
+	const gchar *keys[] = { "NAME=", "VERSION_ID=", "VARIANT=", NULL };
+	const gchar *values[] = { NULL, NULL, NULL, NULL };
+	g_autofree gchar *os_release = NULL;
+	g_auto(GStrv) lines = NULL;
+	g_autoptr(GPtrArray) ids_os = g_ptr_array_new ();
+
+	/* get raw data then parse each line */
+	if (!g_file_get_contents ("/etc/os-release", &os_release, NULL, NULL))
```
Comment (superm1) [inline libfwupd/fwupd-common.c]: I think you should set up a fall back according to freedesktop.

https://www.freedesktop.org/software/systemd/man/os-release.html

File: libfwupd/fwupd-common.c [hunk 2]
```diff
@@ -133,3 +135,115 @@ fwupd_checksum_get_best (GPtrArray *checksums)
...
+{
+	struct utsname name_tmp = { 0 };
+	g_autofree gchar *locale = NULL;
+	g_autofree gchar *os_release = NULL;
+	g_autoptr(GPtrArray) ids = g_ptr_array_new_with_free_func (g_free);
+
+	/* system, architecture and kernel, e.g. "Linux i686 4.14.5" */
+	if (uname (&name_tmp) >= 0) {
+		g_autofree gchar *kernel_version = g_strdup (name_tmp.release);
+		g_strdelimit (kernel_version, "-", '\0');
```
Comment (superm1) [inline libfwupd/fwupd-common.c]: I don't think it's a good idea to strip out distro kernel build number. Sometimes large patches are backported and you wouldn't be able to cross reference distro change log to tell if they are there.

File: libfwupd/fwupd-common.c (hunk 1 above)
Comment (hughsie) [reply libfwupd/fwupd-common.c]: Good catch, fixed.

File: libfwupd/fwupd-common.c (hunk 2 above)
Comment (hughsie) [reply libfwupd/fwupd-common.c]: Makes sense I guess, fixed.

**Pull request merged**